etl/src/energuide/embedded/basement.py

```diff
@@ -104,7 +104,23 @@ class BasementFloor(_BasementFloor):
     }
 
     @classmethod
-    def _from_data(cls, floor: element.Element, construction_type: str, floor_type: FloorType) -> 'BasementFloor':
+    def _from_data(cls,
+                   floor: typing.Optional[element.Element],
+                   construction_type: str,
+                   floor_type: FloorType) -> 'BasementFloor':
+
+        if floor is None:
```

Comment:
This is weird. `construction_type` isn't used if `floor` is None - hints that maybe this should be a different function.

Reply:
Do you think some kind of `_empty_floor` function would be a better idea, that is called instead of this when `floor` in `None`?